File: Facebook/look_and_say.py
import re


def look_and_say(n):
    seq = "1"
    regex = re.compile(r"(\d)\1*")

    for i in range(n - 1):
        next_seq = ""
        for match in regex.finditer(seq):
            n = len(match.group(0))
            char = match.group(0)[0]
            next_seq += f"{n}{char}"
        seq = next_seq

    return seq


print(look_and_say(5))

File: Facebook/look-and-say.js
function flatten(arr) {
  return arr.reduce((x, y) => {
    if (Array.isArray(y)) {
      Array.prototype.push.apply(x, flatten(y));
    } else {
      x.push(y);
    }
    return x;
  }, []);
}

function lookAndSay(n) {
  // O(n + 2m) = O(n + m)
  let seq = "1";
  let re = /(\d)\1*/g;

  for (let i = 0; i < n; i++) {
    // O(n)
    let groups = seq.match(re); // O(m)
    seq = groups.reduce((x, y) => {
      // worst case, O(m)
      return `${x}${y.length}${y[0]}`;
    }, "");
  }

  return seq;
}
